test/ctim/domain/id_test.cljc

```diff
@@ -85,3 +85,15 @@
                                 {:protocol "http"
                                  :hostname "ctia.com"
                                  :path-prefix "/bar"})))))
+
+(deftest test-long-id->id
+  (testing "with hyphens in the entity type"
+    (is (= #ctim.domain.id.CtiaId{:hostname "localhost",
+                                  :short-id "exploit-target-d51dfc7b-df40-46a4-9b06-c396e3dfdbcf",
+                                  :path-prefix nil,
+                                  :port 3001,
+                                  :protocol "http",
+                                  :type "exploit-target"}
+           (id/long-id->id
+            (str "http://localhost:3001/ctia/exploit-target/"
+                 "exploit-target-d51dfc7b-df40-46a4-9b06-c396e3dfdbcf"))))))
```

Comment:
It's only a test, so I'm less worried, but I would extract the short-id portion into a def:
```
(def test-id "exploit-target-d51dfc7b-df40-46a4-9b06-c396e3dfdbcf")

(deftest test-long-id->id
  (testing "with hyphens in the entity type"
    (is (= #ctim.domain.id.CtiaId{:hostname "localhost",
                                  :short-id test-id,
                                  :path-prefix nil,
                                  :port 3001,
                                  :protocol "http",
                                  :type "exploit-target"}
           (id/long-id->id
            (str "http://localhost:3001/ctia/exploit-target/" test-id))))))
```